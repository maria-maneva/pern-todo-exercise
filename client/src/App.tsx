import React, { useEffect } from "react";
import InputTodo from "components/InputTodo";
import ListTodos from "components/ListTodos";
import { getCategories } from "requests";
import { useDataDispatch } from "hooks/useDataDispatch";
import { setCategories } from "store/data.actions";
import { ICategory } from "interfaces";

function App() {
  const dataDispatch = useDataDispatch();

  useEffect(() => {
    (async () => {
      const categories = await getCategories();
      dataDispatch(setCategories(categories as ICategory[]));
    })();
  }, [dataDispatch]);

  return (
    <div className="App container" data-testid="main-container">
      <h1 className="text-center mt-5">PERN (bootstrap) "To Do" List</h1>
      <InputTodo />
      <ListTodos />
    </div>
  );
}

export default App;
