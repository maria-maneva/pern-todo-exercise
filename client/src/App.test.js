import React from "react";
import { render, screen } from "@testing-library/react";
import App from "./App";
import renderer from "react-test-renderer";

test("renders main container", () => {
  render(<App />);
  const root = screen.getByTestId("main-container");
  expect(root).toBeInTheDocument();
});

test("renders input form", () => {
  render(<App />);
  const inputForm = screen.getByTestId("input-form");
  expect(inputForm).toBeInTheDocument();
});

test("renders  todos list", () => {
  render(<App />);

  const todosList = screen.getByTestId("list-todos");
  expect(todosList).toBeInTheDocument();
});

test("renders correctly", () => {
  const view = renderer.create(<App />).toJSON();
  expect(view).toMatchSnapshot();
});
