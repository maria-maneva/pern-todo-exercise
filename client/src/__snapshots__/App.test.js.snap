// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App container"
  data-testid="main-container"
>
  <h1
    className="text-center mt-5"
  >
    PERN (bootstrap) "To Do" List
  </h1>
  <div
    data-testid="input-form"
  >
    <h6
      className="mt-5"
    >
      Add to do:
    </h6>
    <form
      className="d-flex flex-column "
      onSubmit={[Function]}
    >
      <input
        className="form-control"
        data-testid="description-input"
        onChange={[Function]}
        placeholder="Enter description..."
        type="text"
        value=""
      />
      <div
        className="mt-2 mb-2"
        data-testid="category-list"
      >
        <h6>
          Select category
        </h6>
        <div>
          <input
            data-testid="category-input"
            onChange={[Function]}
            placeholder="Enter category divided by comma"
            style={
              Object {
                "border": "none",
                "display": "inline",
                "minWidth": "250px",
                "outline": "none",
              }
            }
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        className="btn btn-primary disabled"
      >
        Add
      </button>
    </form>
  </div>
  <table
    className="table text-start align-middle mt-5"
    data-testid="list-todos"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          Description
        </th>
        <th
          scope="col"
        >
          Categories
        </th>
        <th
          className="text-right"
          scope="col"
        >
          Edit
        </th>
        <th
          className="text-right"
          scope="col"
        >
          Delete
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
